packages/graphql-language-service/src/utils/getVariablesJSONSchema.ts

```diff
@@ -99,14 +141,14 @@ function renderTypeToString(
   return into.join('');
 }
 
-const scalarTypesMap: { [key: string]: JSONSchema6TypeName } = {
-  Int: 'integer',
-  String: 'string',
-  Float: 'number',
-  ID: 'string',
-  Boolean: 'boolean',
+const defaultScalarTypesMap: { [key: string]: JSONSchema6 } = {
+  Int: { type: 'integer' },
+  String: { type: 'string' },
+  Float: { type: 'number' },
+  ID: { type: 'string' },
+  Boolean: { type: 'boolean' },
   // { "type": "string", "format": "date" } is not compatible with proposed DateTime GraphQL-Scalars.com spec
-  DateTime: 'string',
+  DateTime: { type: 'string' },
```

Comment:
is there a reason we don't specify `format: 'date-time'` here?
I think these are compatible:
https://scalars.graphql.org/andimarek/date-time
https://json-schema.org/draft/2020-12/json-schema-validation.html#section-7.3.1

Reply:
After double checking, there is no spec for `Date`.
Also, JSONSchema `date-time` is not fully compatible with the specs
e.g. offset is required

I might be able to come with a regex that follows the specs.